### PR TITLE
SNOW-221819 externalbrowser not working

### PR DESCRIPTION
**Pull request by sfc-gh-stakeda**
### Description
Fixed temporary by using the channels

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: authexternalbrowser.go [hunk 1]
```diff
@@ -189,67 +189,64 @@ func authenticateByExternalBrowser(
...
+			c.Close()
+			encodedSamlResponseChan <- encodedSamlResponse
+			errChan <- errAccept
+		}(conn)
 
-	if tokenErr != nil {
-		return nil, nil, tokenErr
+		encodedSamlResponse = <-encodedSamlResponseChan
+		errFromGoroutine = <-errChan
+		break
```
Comment (sfc-gh-jihuang) [inline authexternalbrowser.go:245]: `authexternalbrowser.go:245:3: the surrounding loop is unconditionally terminated (SA4004)`

**Comment by sfc-gh-jihuang** [comment]
@smtakeda : I am having trouble getting past the Makefile.  I don't know what parameters.json is supposed to be, because I have only been doing externalbrowser authentication with my Okta user up to this point.  Is there an account we are expected to use?

```
go mod vendor
kill -9 $(ps -ewf | grep hang_webserver | grep -v grep | awk '{print $2}') || true
python3 ci/scripts/hang_webserver.py 12345 &
./ci/scripts/test_component.sh
SNOWFLAKE_TEST_ACCOUNT=testaccount
SNOWFLAKE_TEST_DATABASE=testdatabase
SNOWFLAKE_TEST_ROLE=testrole
SNOWFLAKE_TEST_SCHEMA=testschema
SNOWFLAKE_TEST_USER=testuser
SNOWFLAKE_TEST_WAREHOUSE=testwarehouse
HTTP Server Running on PORT 12345..........
panic: failed to create schema. 260008 (08004): failed to connect to db. verify account name is correct. HTTP: 403, URL: https://testaccount.snowflakecomputing.com:443/session/v1/login-request?databaseName=testdatabase&requestId=949aea9e-65fe-4708-9baa-e6027bc236c1&request_guid=e331be17-c551-4001-bede-d46c4aedfe61&roleName=testrole&schemaName=testschema&warehouse=testwarehouse

goroutine 1 [running]:
github.com/snowflakedb/gosnowflake.TestMain(0xc0004e0100)
	/Users/jhuang/workspace/ce-gcp-snowflake-admin/gosnowflake/driver_test.go:151 +0x23f
main.main()
	_testmain.go:445 +0x372
FAIL	github.com/snowflakedb/gosnowflake	6.290s
FAIL
make[1]: *** [test] Error 1
make: *** [cov] Error 2
```

File: authexternalbrowser.go (hunk 1 above)
Comment (sfc-gh-stakeda) [reply authexternalbrowser.go]: ah, fixed in the following changes.

**Comment by sfc-gh-stakeda** [comment]
@sfc-gh-jihuang , ignore those errors as the script is for the CI. You just need to run `go build` under the cmd/externalbrowser directory.

**Comment by sfc-gh-stakeda** [comment]
Btw, I asked @sfc-gh-jbahk to follow up this issue.

File: connection.go [hunk 2]
```diff
@@ -249,7 +249,7 @@ func (sc *snowflakeConn) ExecContext(ctx context.Context, query string, args []d
 		return nil, err
 	}
 
-	var updatedRows int64 = 0
```
Comment (sfc-gh-stakeda) [inline connection.go:252]: The staticcheck raised this.

File: connection.go [hunk 3]
```diff
@@ -487,7 +487,7 @@ func (sc *snowflakeConn) stopHeartBeat() {
 }
 
 func updateRows(data execResponseData) (int64, error) {
-	var count int64 = 0
```
Comment (sfc-gh-stakeda) [inline connection.go:490]: The staticcheck raised this.

**Review by sfc-gh-jbahk — APPROVED** [review]

**Pull request merged**